### PR TITLE
Feature: Update to v3

### PR DESCRIPTION
**Pull request by vaeng**
### General

This is a just one part of the update to v3. It also needs a custom fork of Catch2 with a new reporter, that will directly form the JSON file. Also we need to change the tests in the cpp repository. This merging step should happen first.

If this is successful, the intermediate conversion from XML to JSON with a python script can be scraped.

### Thoughts so far

The previous catch2 reporter was junit, which will not report an any tags, so we cannot group tests by tasks for the concept exercises.
When we change to the XML reporter, this feature is reported and we can take it from there.
Another option would be to keep the  junit reporter and get the tag from the test case name like "<old_test_name>[TASK02]". This would still need the python step to convert from junit-xml to Exercism-json output.

The bad thing about keeping the current setup with junit might be a v4 that can report on more details like test run-time etc. A custom reporter might be the best option.

### TLDR
This PR is one of the parts to move the test runner to v3. It should have three steps:

- Change the run.sh file, to point at the new reporter (this PR)
- Add an exercism reporter to the Catch2 fork
- Change all the tests in the cpp repository to use the new Catch2 version

### CODE REVIEW

**Comment by vaeng** [comment]
As we don't need Python, pip, and the junitparser, we might be able to make a docker image that is smaller.

**Comment by vaeng** [comment]
For the time being, the exercism reporter fork lives at: https://github.com/vaeng/Catch2-with-exercism-reporter

**Comment by vaeng** [comment]
Couldn't get the reporter to work in my favor. So I ditched the idea so far and used the XML reporter with a home brew parser. 
I think maintenance might be easier this way.

#72 is my suggestion to move the runner to v3.
